### ISSUE TITLE
What if Drop references other fields?

### ISSUE THREAD

**Issue by diwic** [Mixthos/self-ref#1]
Hi, I found your crate while considering making something similar.

I was wondering what might happen if you have fields referencing other fields in their `Drop` destructors - won't this be potentially be a use-after-free?

**Comment by Mixthos** [comment]
You're right. I just did some testing and Rust seems to call `drop()` first on the containing struct and then in the order the fields were declared. I can't find a definition of that though, so it's probably undefined behavior.

I'm not sure how to fix this. For now, you should make sure you assign the fields in the order you declared them.
